### ISSUE TITLE
Cluster fails to become 'ready' when existing servicePrincipal is used

### ISSUE THREAD

**Issue by skinny** [Azure/acs-engine#2349]
**Is this a request for help?**:
no
---

**Is this an ISSUE or FEATURE REQUEST?** (choose one):
ISSUE
---

**What version of acs-engine?**:
0.13
---

**Orchestrator and version (e.g. Kubernetes, DC/OS, Swarm)**
Kubernetes 1.9.3

**What happened**:
After deploying a new cluster with prepopulated ClientId & Secret in the apimodel.json and networkPolicy 'calico' neither master or agents became `Ready` but stay in the `NotReady` state

Cluster deployed successfully but failed to schedule or deploy anything because the used credentials did not have the `Contributor` role in the newly created resource group

**What you expected to happen**:
An up and running cluster

**How to reproduce it** (as minimally and precisely as possible):

Deploy a new cluster with prepopulated ClientId & Secret in the apimodel.json with networkPolicy 'calico' (to make it visible from the start)

**Anything else we need to know**:

My suggestion would be to at least show a warning to the user when the used credentials don't have the correct permissions to the resource group. Better (?) would be to try to assign the correct role during deployment.  After all, when you leave it up to acs-engine to create the principal it also assigns the correct role automatically

**Comment by CecileRobertMichon** [comment]
Closing as this a duplicate of #2157 which was fixed with #2297.

**Comment by skinny** [comment]
I also tried this with a build from source (with this change included) and that doesn't seem to fix my issue describe here. The check here https://github.com/vvvrrooomm/acs-engine/blob/5672ee4692daa9cad376c98cdefccdc3d1f80d68/cmd/deploy.go#L205 will prevent this code from executing when you have provided client+secret in your apimodel.json 

I might be missing something here (golang is not my main programming language) so I'm open to corrections :-)

**Comment by CecileRobertMichon** [comment]
@skinny you are right I jumped to conclusions too quickly... Looking at it again #2297 only handles the case where the ServicePrincipleProfile is empty so it doesn't check permissions of the existing SP. #2157 was also about checking for permissions and #2297 only partially addressed it.
